ElectronClient/app/gui/ResourceScreen.tsx

```diff
@@ -0,0 +1,180 @@
+		<a href="#" onClick={ () => props.onToggleSortOrder(s) }>{ props.sortingOrders.includes(s) ? '▾' : '▴' }</a>
+	);
+	return <table style={{ width: '90%' }}>
+		<thead>
+			<tr>
+				<th>Title { sortOrderEngagedMarker('name') }</th>
+				<th>Size { sortOrderEngagedMarker('size') }</th>
+				<th>ID</th>
+				<th>Action</th>
+			</tr>
+		</thead>
+		<tbody>
+			{sortedResources.map((resource: IResource, index: number) =>
+				<tr key={index}>
+					<td>
+						<a href="#" onClick={() => props.onResourceClick(resource)}>{resource.title}</a>
+						{ resource.isOrphan && <i>(Orphan)</i> }
+					</td>
+					<td>{resource.size}</td>
+					<td>{resource.id}</td>
+					<td>
+						<button onClick={ () => props.onResourceDelete(resource) }>Delete</button>
+					</td>
+				</tr>
+			)}
+		</tbody>
+	</table>;
+};
+
+const openResource = (resource: IResource) => {
+	const resourcePath = Resource.fullPath(resource.daoResource);
+	const ok = bridge().openExternal(`file://${resourcePath}`);
+	if (!ok) {
+		bridge().showErrorMessageBox(_('This file could not be opened: %s', resourcePath));
+	}
+};
+
+const deleteResource = (resource: IResource): Promise<any> => Resource.batchDelete([resource.id]);
+
+class ResourceScreenComponent extends React.Component<IProps, IState> {
+	constructor(props: IProps) {
+		super(props);
+		this.state = {
+			resources: undefined,
+			sortingOrders: ['size', 'orphaned'],
+		};
+	}
+
+	async reloadResources() {
+		const noteResources = await NoteResource.modelSelectAll('SELECT * FROM note_resources');
+		const resources = await Resource.modelSelectAll('SELECT * FROM resources');
+		this.setState({ resources: buildResources(resources, noteResources) });
+	}
+
+	componentDidMount() {
+		this.reloadResources();
+	}
+
+	render() {
+		const style = this.props.style;
+		const theme = themeStyle(this.props.theme);
+		const headerStyle = Object.assign({}, theme.headerStyle, { width: style.width });
+		return <>
+			<Header style={headerStyle} />
+			<div style={{ ...style, margin: '20px', overflow: 'scroll' }}>
+				{!this.state.resources && <div>
+					No Resources!
+				</div>
+				}
+				{this.state.resources && <ResourceTable
+					resources={ this.state.resources }
+					sortingOrders={ this.state.sortingOrders }
+					onToggleSortOrder={ (sortOrder) => {
+						if (this.state.sortingOrders.includes(sortOrder)) {
+							this.setState({ sortingOrders: this.state.sortingOrders.filter(s => s !== sortOrder) });
+						} else {
+							this.setState({ sortingOrders: [ ...this.state.sortingOrders, sortOrder ] });
+						}
+					} }
+					onResourceClick={ (resource) => {
+						openResource(resource);
+					}}
+					onResourceDelete={ (resource) => {
+						deleteResource(resource)
+							.catch(error => {
+								bridge().showErrorMessageBox(_('Resource deletion failed: %s', error));
+							})
+							.finally(() => {
+								this.reloadResources();
+							});
+					} }
+				/>}
```

Comment:
I would prefer if all the event handlers were in separate class methods. That will also mean less unnecessary renderings.

Reply:
Sure. I never heard before that putting it as a class method reduced re-renderings, do you happen to have a documentation for that?